content/developer/tutorials/server_framework_101/06_security.rst

```diff
@@ -348,10 +348,128 @@ access to property records.
 Separate company data
 =====================
 
-tmp
+In enterprise environments, organizations often need to manage multiple distinct entities within the
+same system. This approach allows different branches, subsidiaries, franchises, or even completely
+different companies to operate independently while sharing common resources and functionalities.
+
+In Odoo, the **multi-company** feature enables managing multiple companies within a single database.
+Each company can have its own configuration and data, while still allowing users to access data from
+multiple companies. This is implemented through several key mechanisms:
+
+- **Company field**: Adding a `company_id` many-to-one field to a model allows linking its records
+  to a specific company (represented by the generic `res.company` model). Records can be:
+
+  - **Company-specific**: When `company_id` has a value, making the record belong to one company.
+  - **Company-shared**: When `company_id` is empty, making the record accessible across all
+    companies.
+
+- **Company-dependent fields**: The `company_dependent=True` attribute set on a field creates a
+  separate value for each company. The values are stored in a JSON object in the database and the
+  right value is automatically retrieved based on the current company.
+- **Company context**: The `with_company(company)` model method changes the company context when
+  accessing data like company-dependent fields, allowing to retrieve values from a specific
+  company's perspective.
```

Comment:
retrieve values **and trigger flows** from a specific company perspective ?

It's not only about retrieving values (though not sure how to clearly explain it here).
In the case where a users selects company A and B (A as main) but triggers a flow on B, it's a good practice to trigger the flow in company B to make sure nothings breaks (or is wrongly configured).

Reply:
Ok, I'll accept this one, but because there is a "trigger business workflow" section in the previous chapter :grin: